app/controllers/CheckStateActor.scala

```diff
@@ -16,27 +16,40 @@
 
 package controllers
 
+import java.time.LocalDateTime
+
 import akka.actor.Actor
 import akka.pattern.{ask, pipe}
 import akka.util.Timeout
+import config.FrontendAppConfig
+import javax.inject.Inject
 import repositories.SessionRepository
 import services.{FileUploadService, FileUploadState, FileUploaded, UploadFile}
 
-import java.time.LocalDateTime
-import javax.inject.Inject
-import scala.concurrent.duration.DurationInt
+import scala.concurrent.duration.{FiniteDuration, SECONDS}
 import scala.concurrent.{ExecutionContext, Future}
 
+trait CheckStateSupport {
+  val appConfig: FrontendAppConfig
+  implicit val checkStateActorTimeout = Timeout(appConfig.fileUploadTimeout.plus(FiniteDuration(5, SECONDS)))
```

Comment:
why do we add 5 seconds on?

Reply:
Because this is the timeout for the actor to return a response.  It was previously the same as the file upload timeout (10 secs) and I was occasionally getting an exception where the actor failed to respond quickly enough.
You will notice that the actor used to have it's own implicit timeout of 30 seconds - I think that was the timeout used when the actor posted a message to itself.